x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/type/DataTypeConversion.java

```diff
@@ -87,16 +89,32 @@ public static DataType commonType(DataType left, DataType right) {
                 return right;
             }
         }
-        if (left == DATETIME) {
+        if (left == TIME) {
```

Comment:
Considering `TIME` is a no-so-used type, I would expect handling of `DATETIME` first and then `TIME` not vice-versa.

Reply:
+1
Will do it in a followup everywhere if you don't mind.